### PR TITLE
[migration phase 1] PodFitsResources as framework plugin

### LINKED ISSUE #83630: [migration phase 1] PodFitsResources as framework plugin
**What would you like to be added**:

Add a filter plugin that calls into PodFitsResources predicate, example #83460

**Why is this needed**:

Part of  #83554

/sig scheduling
/help
/priority important-soon

### PR DESCRIPTION
**Pull request by wgliang**
**What type of PR is this?**

 /kind cleanup
/sig scheduling
/priority important-soon

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #83630

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
[migration phase 1] PodFitsResources as framework plugin
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

**Comment by wgliang** [comment]
@ahg-g Ready for review. Thanks.

**Comment by ahg-g** [comment]
Thanks Leon, lets names this ```NodeResources```

/cc @alculquicondor @Huang-Wei

File: pkg/scheduler/framework/plugins/noderesources/node_resources.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+const Name = "NodeResources"
+
+// Name returns name of the plugin. It is used in logs, etc.
+func (pl *NodeResources) Name() string {
+	return Name
+}
+
+// Filter invoked at the filter extension point.
+func (pl *NodeResources) Filter(_ *framework.CycleState, pod *v1.Pod, nodeInfo *nodeinfo.NodeInfo) *framework.Status {
+	_, reasons, err := predicates.PodFitsResources(pod, nil, nodeInfo)
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/noderesources/node_resources.go]: comment that the metadata is not needed.

**Comment by wgliang** [comment]
@ahg-g Rebased and Ready for review. Thanks.

File: pkg/scheduler/framework/plugins/noderesources/node_resources.go (hunk 1 above)
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/noderesources/node_resources.go]: We need to read the metadata from CycleState and pass it to PodFitsResources, it is used inside the function.

Comment (wgliang) [reply pkg/scheduler/framework/plugins/noderesources/node_resources.go]: done

File: pkg/scheduler/framework/plugins/migration/utils.go [hunk 2]
```diff
@@ -77,6 +77,12 @@ func (p *PredicatesStateData) Clone() framework.StateData {
 	}
 }
 
+func (p *PredicatesStateData) AddPod() framework.StateData {
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/migration/utils.go]: what is this for?

Comment (wgliang) [reply pkg/scheduler/framework/plugins/migration/utils.go]: Sorry, I forgot to delete it. And done.

**Pull request merged**